### PR TITLE
TRC-25- Add `Icon` Component

### PR DESCRIPTION
**Pull request by mmcbride2929**
TRC-25 - feat: add icon component

TL/DR
This story creates an icon component story & an icon component. 

Overview of Changes

`package.json`
- add svgr 
- created script to convert our svg's within `src/assets` to react components

`README.md`
- updated the readme on how and why we are using the package `svgr`

`src/asssets`
- add icon svgs

`Icon.stories.tsx`
- create icon story + variants of each icon + global styles for them
- eliminated use of no longer needed args

`Icon.tsx`
- create icon component that takes children as a prop, and renders the svg icon passed to it.
- removed style prop that was un-needed
- used react composition to create more a readable code structure
- added an enumerator that allows us to reference one source when passing our icons a color

### CODE REVIEW

File: apps/web/package.json [hunk 1]
```diff
@@ -11,11 +11,13 @@
     "lint": "eslint src --fix",
     "clean": "rm -rf build",
     "storybook": "start-storybook -p 6006 -s public",
-    "build-storybook": "build-storybook -s public"
+    "build-storybook": "build-storybook -s public",
+    "build:svgs": "svgr --icon --out-dir src/assets"
```
Comment (wijohnst) [inline apps/web/package.json:15]: This is smart. I hadn't thought about adding this as a build step.

Comment (wijohnst) [reply apps/web/package.json:15]: Perhaps update the readme to talk about when this should be run?

File: apps/web/src/components/Icon/Icon.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,8 @@
+export {};
+// import React from 'react';
```
Comment (wijohnst) [inline apps/web/src/components/Icon/Icon.spec.tsx:2]: Can you add a comment about when this will be updated and link it to an issue? And if we don't think we will write tests for this component this file can be removed.

File: apps/web/src/components/Icon/Icon.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+   * child icon SVG component passed to the Icon component
+   */
+  children: JSX.Element;
+  /**
+   * global css styles applied to all of our Icon components
+   */
+  style?: React.CSSProperties;
+}
+
+const sharedStyles = {
```
Comment (wijohnst) [inline apps/web/src/components/Icon/Icon.stories.tsx:22]: Why are you passing this as a prop to all instances of the `Icon` component? Can't it just live in that scope and save you from passing a prop?

File: apps/web/src/components/Icon/Icon.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+  width: '100%',
+  height: '100%',
+  margin: '5px',
+};
+
+const Template = (args: IconArgs): JSX.Element => <Icon {...args}></Icon>;
+
+export const Reddit = (args: IconArgs) => <Template {...args} />;
```
Comment (wijohnst) [inline apps/web/src/components/Icon/Icon.stories.tsx:30]: When creating a story I like to always have a 'Default' story that includes all of the props that will be the same across my different components. So for this I'd pass an svg in the `children` prop and `sharedStyles to `style`. Then for very other story I can spread in the `Default.args` object, saving me some lines. So it would look something like:

```tsx
export const Default = (args: IconArgs) => <Template {...args} />;
Default.args = {
children: <DefaultIcon color='#ff2800' />,
style: sharedStyles,
}

export const Reddit = (args: IconArgs) => <Template {...args />;
Reddit.args = {
  ...Default.args,
  children: <RedditIcon color="ff2800" />,
}
```

File: apps/web/src/components/Icon/Icon.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+Reddit.args = {
+  children: <RedditIcon color="#ff2800" />,
+  style: sharedStyles,
+};
+
+export const User = (args: IconArgs) => <Template {...args} />;
```
Comment (wijohnst) [inline apps/web/src/components/Icon/Icon.stories.tsx:36]: I think the more semantic way of passing `children` to a component is to use the nested syntax vs. an inline prop:

```ts
export const User = (args: IconArgs) => (
  <Template {...args}>
    <UserIcon />,
  </Template>
)

Comment (wijohnst) [reply apps/web/src/components/Icon/Icon.stories.tsx:36]: This is the same thing as explicitly declaring a `children` prop, and it will be more readable to other devs.

File: apps/web/src/components/Icon/Icon.stories.tsx [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+User.args = {
+  children: <UserIcon />,
+  style: sharedStyles,
+};
+
+export const Home = (args: IconArgs) => <Template {...args} />;
+Home.args = {
+  children: <HomeIcon />,
```
Comment (wijohnst) [inline apps/web/src/components/Icon/Icon.stories.tsx:44]: Why is the Reddit icon the only SVG that accepts a `color` prop?

File: apps/web/src/components/Icon/Icon.stories.tsx [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+  width: '100%',
+  height: '100%',
+  margin: '5px',
+};
+
+const Template = (args: IconArgs): JSX.Element => <Icon {...args}></Icon>;
+
+export const Reddit = (args: IconArgs) => <Template {...args} />;
+Reddit.args = {
+  children: <RedditIcon color="#ff2800" />,
```
Comment (wijohnst) [inline apps/web/src/components/Icon/Icon.stories.tsx:32]: This implementation of the `color` prop is really tightly coupled. What happens if one day we want to change the color of all `RedditIcon` components to `#ff2801`? How can we accomplish this but have it reference a single source of truth?

File: apps/web/src/components/Icon/Icon.tsx [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+/**
+ * Renders an icon component.
+ *
+ * @param {Object} props - The component props.
+ * @param {React.ReactElement} props.children - The child element to render within the icon.
+ * @param {React.CSSProperties} props.style - The inline styles to apply to the icon container element.
+ * @returns {JSX.Element} - A JSX element representing the icon component.
+ */
+const Icon = ({ children, style }: Props) => {
+  return <div style={style}>{children}</div>;
```
Comment (wijohnst) [inline apps/web/src/components/Icon/Icon.tsx:18]: Why isn't the style coming from the stylesheet for the `Icon` component?

**Review by wijohnst — COMMENTED** [review]
Left some comments.

File: apps/web/src/components/Icon/Icon.tsx [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+import './Icon.scss';
+
+export enum iconColor {
+  Default = 'black',
+  Orange = '#ff2800',
+}
+
+interface Props {
+  children: React.ReactElement;
+  colorProp?: iconColor;
```
Comment (wijohnst) [inline apps/web/src/components/Icon/Icon.tsx:11]: I think you end up with a more readable API if you keep the prop name and reference string the same. 

```suggestion
  colorProp?: colorProp;
```

File: apps/web/src/components/Icon/Icon.tsx [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+
+/**
+ * Renders an icon component.
+ *
+ * @param {Object} props - The component props.
+ * @param {React.ReactElement} props.children - The child element to render within the icon.
+ * @returns {JSX.Element} - A JSX element representing the icon component.
+ */
+const Icon = ({ children, colorProp }: Props) => {
+  return <div style={{ color: colorProp }}>{children}</div>;
```
Comment (wijohnst) [inline apps/web/src/components/Icon/Icon.tsx:22]: I think it's best practice to add a default value for optional props, especially when using Storybook. This way the `default` story only passes props that are required. 

```suggestion
  return <div style={{ color: colorProp = ColorsEnum['some-color']}}>{children}</div>;
```

**Review by wijohnst — COMMENTED** [review]
Left some comments. Also, make sure you open this PR up / take it out of draft status.

File: apps/web/src/components/Icon/Icon.tsx [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+}
+
+/**
+ * Renders an icon component.
+ *
+ * @param {Object} props - The component props.
+ * @param {React.ReactElement} props.children - The child element to render within the icon.
+ * @returns {JSX.Element} - A JSX element representing the icon component.
+ */
+const Icon = ({ children, colorProp }: Props) => {
```
Comment (wijohnst) [inline apps/web/src/components/Icon/Icon.tsx:21]: Could you not just set a default prop?
```suggestion
const Icon = ({ children, colorProp = iconColor.Default}: Props) => {
```

**Review by wijohnst — APPROVED** [review]
Good job hanging tough on this one. The first big PR is always the hardest one. It'll get easier from here. Great job on this! 👏
